[AY1920S1-CS2113T-T12-1] Optix

NicholasLiu97 opened the pull request:
@OungKennedy @CheeSengg @TianchangLiao

CheeSengg committed:
Team repo set-up

CheeSengg committed:
Updated base code

CheeSengg committed:
Updated: DateFormatter class to accept one single date time format

CheeSengg committed:
Merge branch 'gradle'

CheeSengg committed:
Added: Gradle and CheckStyle

CheeSengg committed:
Merge pull request #3 from CheeSengg/master

Updated: DateFormatter class to accept one single date time format

CheeSengg committed:
A-Jar

CheeSengg committed:
Added: DateMap and TimeMap for Schedule viewing and detect anomalies

OungKennedy committed:
add reminders function. List all deadlines.edited dukeresponse file as well.

CheeSengg committed:
Fixed: edge cases for DateMap, Added: JUnit for DateMap

OungKennedy committed:
first changes for snooze command.

OungKennedy committed:
merged parsesnooze branch.(7380ce8)

OungKennedy committed:
snooze, findfreetime, reminders working.

CheeSengg committed:
B-DetectAnomalies

OungKennedy committed:
Merge tag 'B-Snooze'

snooze function for events and deadlines.

OungKennedy committed:
first draft for findfreetimes.

OungKennedy committed:
resolving conflicts.event class changed(added functions).

CheeSengg committed:
B-ViewSchedule

OungKennedy committed:
first stages of adding findfreetimes.

NicholasLiu97 committed:
B-DoAfterTasks, to debug date reading

NicholasLiu97 committed:
Merge pull request #7 from NicholasLiu97/master

B-DoAfterTasks, to debug date reading

CheeSengg committed:
Comply with coding standards

OungKennedy committed:
non meaningful commit.

OungKennedy committed:
merge without pulling. changes not meaningful.

OungKennedy committed:
merged from master.

OungKennedy committed:
removed getDate, as it is similar to getDateTime. getDateTime remains.

OungKennedy committed:
non meaningful commit.

OungKennedy committed:
merged from master.

CheeSengg committed:
Merge pull request #9 from OungKennedy/dummybranch

CheeSengg committed:
Morph into Opera

OungKennedy committed:
Merge pull request #10 from CheeSengg/opera-morph

Morph into Opera

CheeSengg committed:
Added: Storage Class and ListCommand Class

CheeSengg committed:
Added: PostponeCommand, to postpone event date

OungKennedy committed:
Added delete commands (#12)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* Morph into Opera

* Added: Storage Class and ListCommand Class

* Added: PostponeCommand, to postpone event date

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

OungKennedy committed:
Merge branch 'opera-morph' into master

OungKennedy committed:
Added delete commands (#12) (#13)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* Morph into Opera

* Added: Storage Class and ListCommand Class

* Added: PostponeCommand, to postpone event date

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

NicholasLiu97 committed:
Create Images file

CheeSengg committed:
Opera morph (#14)

* Added: PostponeCommand, to postpone event date

* Added delete commands (#12)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* Morph into Opera

* Added: Storage Class and ListCommand Class

* Added: PostponeCommand, to postpone event date

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

* Create Images file

* Added ListShowCommand

CheeSengg committed:
Refactored Code into ShowMap<LocalDate, Theatre>

CheeSengg committed:
Refactored Code into ShowMap<LocalDate, Theatre>

CheeSengg committed:
Added ViewSeatsCommand

CheeSengg committed:
Added SellSeatsCommand

OungKennedy committed:
Add files via upload

Image of OungKennedy

OungKennedy committed:
Create AboutUs.adoc

added roles and responsibilities. Missing 3 images.

OungKennedy committed:
Update AboutUs.adoc

OungKennedy committed:
Update AboutUs.adoc

changed formatting

OungKennedy committed:
Merge branch 'opera-morph' into opera-morph

CheeSengg committed:
Merge pull request #16 from CheeSengg/opera-morph

Merge into team repo

CheeSengg committed:
Updated AboutUs

OungKennedy committed:
Update README.md

added 1 feature description. more to come.

OungKennedy committed:
fixed bugs.

OungKennedy committed:
Update README.md

OungKennedy committed:
refactored delete-one to delete as per discussion.

OungKennedy committed:
added javafx functionality.

CheeSengg committed:
Opera morph (#19)

* Updated SellSeatsCommand

* Updated Storage to store Seats sold

OungKennedy committed:
Opera morph (#21)

* fixed bugs.

* refactored delete-one to delete as per discussion.

* added javafx functionality.

* added junit test for AddCommand.

* added junit test for delete-all command.

* added j-unit test for delete(one) command.

* add tests for list command.

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Update AboutUs.adoc

NicholasLiu97 committed:
Update AboutUs.adoc

CheeSengg committed:
Added: HelpCommand Class

CheeSengg committed:
Refactored UI reponse

CheeSengg committed:
Added OptixDateFormatter, refactored all dateFormatting methods in Command

NicholasLiu97 committed:
Create UserGuide.adoc

NicholasLiu97 committed:
Create DeveloperGuide.adoc

NicholasLiu97 committed:
rename README.md to README.adoc

NicholasLiu97 committed:
Add GUI screenshot via upload

NicholasLiu97 committed:
Update README.adoc

NicholasLiu97 committed:
Update README.adoc

NicholasLiu97 committed:
Update README.adoc for v1.1

NicholasLiu97 committed:
Delete README.md from docs folder

NicholasLiu97 committed:
Update README.adoc for v1.1

NicholasLiu97 committed:
Added user stories in DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Added use cases in DeveloperGuide.adoc

NicholasLiu97 committed:
Added use cases in DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Test format of DeveloperGuide.adoc

NicholasLiu97 committed:
Test format for DeveloperGuide.adoc

NicholasLiu97 committed:
Added use cases in DeveloperGuide.adoc

NicholasLiu97 committed:
Added functional requirements to DeveloperGuide.adoc

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Added glossary to DeveloperGuide.adoc

NicholasLiu97 committed:
Added introduction to UserGuide.adoc

NicholasLiu97 committed:
Update UserGuide.adoc

NicholasLiu97 committed:
Update UserGuide.adoc

NicholasLiu97 committed:
Update UserGuide.adoc

NicholasLiu97 committed:
Added GUI image

NicholasLiu97 committed:
Update picture of GUI

NicholasLiu97 committed:
Added GUI picture

CheeSengg committed:
Updated: HelpCommand, BugFix: AddCommand, PostponeCommand, DeleteOneCommand (#23)

* Added: HelpCommand Class

* Refactored UI reponse

* Added OptixDateFormatter, refactored all dateFormatting methods in Command

* Added: Exception Handling Classes

* Updated: HelpCommand, BugFix: DeleteOneCommand, PostponeCommand and AddCommand

NicholasLiu97 committed:
Merge branch 'opera-morph'

NicholasLiu97 committed:
deleted unnecessary files

CheeSengg committed:
Merge pull request #61 from NicholasLiu97/master

Merge opera-morph into master, deleted duke

CheeSengg committed:
Remove performances in the past

CheeSengg committed:
Refactored Commands into packages

CheeSengg committed:
Updated JavaDoc

CheeSengg committed:
Bug Fix for Invalid date input.

OungKennedy committed:
Added j-unit tests (#66)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

* fixed bugs.

* refactored delete-one to delete as per discussion.

* added javafx functionality.

* added junit test for AddCommand.

* added junit test for delete-all command.

* added j-unit test for delete(one) command.

* add tests for list command.

* fixed small bug in viewseatscommand, added viewseatscommandtest.

* pulled new changes from organization repo. added junit test for listshowcommand.

* dded postpone command junit test.

* added sellseatcommandtest. unable to pass test where invalid seat number is given.

CheeSengg committed:
Added: EditCommand to update show name

CheeSengg committed:
Junit Test for EditCommand

CheeSengg committed:
Added Archive for past performances. (#68)

* Remove OptixResponse

* Refactored Files, Changed ShowMap to Model, Added Archive for past performances

OungKennedy committed:
Added travis (#69)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

* fixed bugs.

* refactored delete-one to delete as per discussion.

* added javafx functionality.

* added junit test for AddCommand.

* added junit test for delete-all command.

* added j-unit test for delete(one) command.

* add tests for list command.

* fixed small bug in viewseatscommand, added viewseatscommandtest.

* pulled new changes from organization repo. added junit test for listshowcommand.

* dded postpone command junit test.

* added sellseatcommandtest. unable to pass test where invalid seat number is given.

* added travis.yml

* renamed travis.yml to .travis.yml

* cleaned up a little of the code according to checkstyletest standards. Some standards have to be changed according to our needs.

* code adheres to checkstyle guidelines. added assertions.

* cleaned code. Passes checkstyle test.

* bug fix to postpone command test.

OungKennedy committed:
Aliasing  (#71)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

* fixed bugs.

* refactored delete-one to delete as per discussion.

* added javafx functionality.

* added junit test for AddCommand.

* added junit test for delete-all command.

* added j-unit test for delete(one) command.

* add tests for list command.

* fixed small bug in viewseatscommand, added viewseatscommandtest.

* pulled new changes from organization repo. added junit test for listshowcommand.

* dded postpone command junit test.

* added sellseatcommandtest. unable to pass test where invalid seat number is given.

* added travis.yml

* renamed travis.yml to .travis.yml

* cleaned up a little of the code according to checkstyletest standards. Some standards have to be changed according to our needs.

* code adheres to checkstyle guidelines. added assertions.

* cleaned code. Passes checkstyle test.

* bug fix to postpone command test.

* added aliases for the commands.

CheeSengg committed:
updated photo, storage and .gitignore

CheeSengg committed:
Added ListDateCommand

CheeSengg committed:
Added: ListDateCommandTest

CheeSengg committed:
Refactored Commands

CheeSengg committed:
Handled Checkstyle and Build report

OungKennedy committed:
Command Parser  (#76)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

* fixed bugs.

* refactored delete-one to delete as per discussion.

* added javafx functionality.

* added junit test for AddCommand.

* added junit test for delete-all command.

* added j-unit test for delete(one) command.

* add tests for list command.

* fixed small bug in viewseatscommand, added viewseatscommandtest.

* pulled new changes from organization repo. added junit test for listshowcommand.

* dded postpone command junit test.

* added sellseatcommandtest. unable to pass test where invalid seat number is given.

* added travis.yml

* renamed travis.yml to .travis.yml

* cleaned up a little of the code according to checkstyletest standards. Some standards have to be changed according to our needs.

* code adheres to checkstyle guidelines. added assertions.

* cleaned code. Passes checkstyle test.

* bug fix to postpone command test.

* added aliases for the commands.

* version 2 of CommandParser. aliasing done differently. missing tests.

* this commit passes checkstyletest.

NicholasLiu97 committed:
Added ViewProfitCommand (#77)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

CheeSengg committed:
Refactor model (#78)

* removed buyerName for Seat Class

* removed cost in Theatre Class

* Refactored Theatre Class

NicholasLiu97 committed:
Added support to view monthly profit (#79)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

NicholasLiu97 committed:
Add files via upload

CheeSengg committed:
Refactored AddCommand

CheeSengg committed:
Refactored EditCommand

CheeSengg committed:
Refactored ListCommand

CheeSengg committed:
Refactored PostponeCommand

CheeSengg committed:
Refactored ListShowCommand

CheeSengg committed:
Refactored ListDateCommand and minor bug fix

CheeSengg committed:
Refactored DeleteOneCommand

CheeSengg committed:
Refactored DeleteAllCommand

CheeSengg committed:
Refactored SellSeatCommand

CheeSengg committed:
Refactored ViewSeatsCommand

CheeSengg committed:
Removed ShowHistoryMap

CheeSengg committed:
Updated HelpCommand

CheeSengg committed:
Bug Fix for Storage

akshayeshenoi commented on src/main/java/Optix.java:
```diff
@@ -0,0 +1,69 @@
+    private Storage storage;
+
+    /**
+     * Set up the storage, ui, and list of shows.
```
JavaDoc comments usually begin like "sets up the storage.." instead of "set up the storage..". Try to follow this convention across your project.

akshayeshenoi commented on src/main/java/Optix.java:
```diff
@@ -0,0 +1,69 @@
+     * initialize Optix object.
+     * path to the save file is defined in filePath.
+     *
+     * @param args is empty?
```
Try not to use comments like these. As a developer reading this code, I don't what that is supposed to mean. We should try to keep the code as clear and crisp as possible.

akshayeshenoi commented on src/main/java/Optix.java:
```diff
@@ -0,0 +1,69 @@
+     * To boot up the software.
+     */
+    public void run() {
+
```
While it is a good habit to add an empty line between logical sets of statements, try not to add empty lines in seemingly random places. It increases the overall height of the code and violates our coding standards.

akshayeshenoi commented on src/main/java/optix/commands/HelpCommand.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
If all you are doing here is call the `super` isExit() method, you can choose to skip it altogether, right?

akshayeshenoi commented on src/main/java/optix/commands/Command.java:
```diff
@@ -0,0 +1,24 @@
+import optix.commons.Storage;
+import optix.ui.Ui;
+
+public abstract class Command {
```
I am not sure why this class was created as `abstract`.  
The `quit` variable seems to not serve any purpose and `isExit()` always returns false. Would it not be a better idea to use an interface instead, since there doesn't seem to be any shareable code here?

akshayeshenoi commented on src/main/java/optix/commands/HelpCommand.java:
```diff
@@ -0,0 +1,106 @@
+
+    private static final String ADD_MENU = "To add a new show:                            "
+                                           + "add SHOW_NAME|SCHEDULED_DATE|PRICE|SEATS_BASE_PRICE\n";
+
```
Avoid these newlines (don't add newlines between logically similar variables)

akshayeshenoi commented on src/main/java/optix/commands/parser/AddAliasCommand.java:
```diff
@@ -0,0 +1,58 @@
+    private HashMap<String, String> commandAliasMap;
+
+    /**
+     * Command to add a new alias to the command alias map.
```
"Adds new alias.."

akshayeshenoi commented on src/main/java/optix/commands/seats/SellSeatCommand.java:
```diff
@@ -0,0 +1,132 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class SellSeatCommand extends Command {
```
Very well written file!

akshayeshenoi commented on src/main/java/optix/commands/shows/DeleteOneCommand.java:
```diff
@@ -0,0 +1,55 @@
+            if (showToDelete != null && showToDelete.hasSameName(showName)) {
+                shows.remove(showLocalDate, showToDelete);
+                model.setShows(shows);
+                message.append(String.format(MESSAGE_SUCCESSFUL, showName, showDate));
```
Stick to either `String.format()` or normal string concatenation (using the `+` operator). Don't alternate between both across the codebase.

akshayeshenoi commented on src/main/java/optix/commands/shows/ListShowCommand.java:
```diff
@@ -0,0 +1,60 @@
+    public boolean isExit() {
+        return super.isExit();
+    }
+
```
Avoid empty lines!

akshayeshenoi commented on src/main/java/optix/commands/shows/ViewMonthlyCommand.java:
```diff
@@ -0,0 +1,199 @@
+import java.time.LocalDate;
+import java.util.Map;
+
+public class ViewMonthlyCommand extends Command {
```
Well written!

akshayeshenoi commented on src/main/java/optix/commons/Model.java:
```diff
@@ -0,0 +1,35 @@
+    private ShowMap shows = new ShowMap();
+
+    /**
+     * The Optix model.
```
This needs to be more descriptive. It does not convey the purpose of this class.

akshayeshenoi commented on src/main/java/optix/commons/model/Seat.java:
```diff
@@ -0,0 +1,69 @@
+    private boolean isBooked;
+
+    /**
+     * the seat object.
```
When writing javadoc, ask yourself if the comment is adding any more information than what the reader might infer regardless of it. Try explaining why it exists.

akshayeshenoi commented on src/main/java/optix/commons/model/Theatre.java:
```diff
@@ -0,0 +1,281 @@
+        }
+    }
+
+    private int getCol(String col) {
```
You should perform a character to int conversion rather than using a switch case here.

akshayeshenoi commented on src/main/java/optix/commons/model/Theatre.java:
```diff
@@ -0,0 +1,281 @@
+        return message;
+    }
+
+    private int getRow(String row) {
```
What if you saved these characters in an array and use an `indexOf()` call to get the value?

akshayeshenoi commented on src/test/java/Optix/commands/SellSeatCommandTest.java:
```diff
@@ -0,0 +1,52 @@
+    @Test
+    void execute() {
+        filePath.deleteOnExit();
+        AddCommand addTestShow1 = new AddCommand("Test Show 1", "5/5/2020", 20);
```
You are allowed to use underscores in variable names in unit tests. You could can separate the '1' using an underscore. If you are testing for a specific case, you can even express that in the variable name. Refer our coding standards from the module website.

akshayeshenoi reviewed (commented):
You have written good quality code overall. 

Bear in mind that my comments might be tagged to particular lines but should be applied everywhere. Also certain things like string formatting are not consistent across the codebase. This may be because different people prefer different styles. There is no right way, but try to establish a convention and follow it diligently.

Keep up the good work!

CheeSengg committed:
Enhanced DeleteCommand to allow bulk deletion

CheeSengg committed:
Enhanced AddCommand for bulk addition

CheeSengg committed:
Updated HelpCommand

CheeSengg committed:
Added MainWindow and Dialog Box logic

CheeSengg committed:
ChatBox displays CLI message

CheeSengg committed:
Display closes on command

CheeSengg committed:
Listing of Shows

CheeSengg committed:
CheckStyle

CheeSengg committed:
GUI SeatDisplay

CheeSengg committed:
Shows Button and Archive Button

OungKennedy committed:
Alias handling bug fixes. (#91)

* further attempt to fix code from different branches.

* cherry picked from B-Reminders.

* continued cherry picking.

* merging different code.

* merged stuff.

* removed getDate, as it is similar to getDateTime. getDateTime remains.

* added delete-one and delete-all commands.

* added delete-one and delete-all commands.

* fixed bugs.

* refactored delete-one to delete as per discussion.

* added javafx functionality.

* added junit test for AddCommand.

* added junit test for delete-all command.

* added j-unit test for delete(one) command.

* add tests for list command.

* fixed small bug in viewseatscommand, added viewseatscommandtest.

* pulled new changes from organization repo. added junit test for listshowcommand.

* dded postpone command junit test.

* added sellseatcommandtest. unable to pass test where invalid seat number is given.

* added travis.yml

* renamed travis.yml to .travis.yml

* cleaned up a little of the code according to checkstyletest standards. Some standards have to be changed according to our needs.

* code adheres to checkstyle guidelines. added assertions.

* cleaned code. Passes checkstyle test.

* bug fix to postpone command test.

* added aliases for the commands.

* version 2 of CommandParser. aliasing done differently. missing tests.

* this commit passes checkstyletest.

* added list-alias

* made changes to parser.

* incomplete changes to the parser class.

* fixed bugs in alias handling. Problem was that there was no need to escape pipe.\\| vs |

* this commit passes checkstyletest.

NicholasLiu97 committed:
Added reassign command (#92)

* Refactored AddCommand

* Refactored EditCommand

* Refactored ListCommand

* Refactored PostponeCommand

* Refactored ListShowCommand

* Refactored ListDateCommand and minor bug fix

* Refactored DeleteOneCommand

* Refactored DeleteAllCommand

* Refactored SellSeatCommand

* Refactored ViewSeatsCommand

* Removed ShowHistoryMap

* Updated HelpCommand

* Bug Fix for Storage

* Enhanced DeleteCommand to allow bulk deletion

* Enhanced AddCommand for bulk addition

* Updated HelpCommand

* added JUnit test for ViewMonthlyCommand

* Added Junit test for ViewProfitCommand

* Minor debugging
- ViewMonthlyCommand
- changed tier number for seats

* added reassign seats branch
-buggy

* Fixed bug for reassign method

OungKennedy committed:
merged CheeSengg's code.

OungKennedy committed:
Merge branch 'CheeSengg-javafx'

CheeSengg committed:
Bug Fix for merge

OungKennedy committed:
refactored the commands for better exception handling.

CheeSengg committed:
CheckStyle and Coding Standards

CheeSengg committed:
Updated AddCommandTest

CheeSengg committed:
CheckStyleTest

CheeSengg committed:
SeatTest and ShowTest

CheeSengg committed:
Added TheatreTest

CheeSengg committed:
Added ShowMapTest

OungKennedy committed:
added junit tests for add, delete edit commands.

CheeSengg committed:
Updated User Guide

CheeSengg committed:
inserted diagram

CheeSengg committed:
Fixed Bugs on UserGuide

CheeSengg committed:
Updated UI.png

OungKennedy committed:
added junit test for list list-date command.

CheeSengg committed:
Junit (#102)

* updated seats and shows packages test

* Added FinanceEntry. Updated clickable tabs and Model for GUI components

* Added TabCommand to switch between tab

* Updated Junit test for finance Commands package

OungKennedy committed:
Added junit tests, edited ug (#104)

* added junit tests for add, delete edit commands.

* added junit test for list list-date command.

* resolved conflicts.

* updated user guide. Improved formatting to be clearer using MarkDown.

NicholasLiu97 committed:
Updated view-monthly command (#105)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

NicholasLiu97 committed:
Updated UserGuide.adoc (#108)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

* Updated jar version
Updated user guide
- tips and notes
- content page for navigation
- highlighting of commands and parameters
- minor grammar errors

OungKennedy committed:
Edited UG (#109)

* added junit tests for add, delete edit commands.

* added junit test for list list-date command.

* resolved conflicts.

* updated user guide. Improved formatting to be clearer using MarkDown.

* user guide update for alias related commands.

NicholasLiu97 committed:
Updated UG (#110)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

* Updated jar version
Updated user guide
- tips and notes
- content page for navigation
- highlighting of commands and parameters
- minor grammar errors

NicholasLiu97 committed:
Add sequence diagram for view monthly

NicholasLiu97 committed:
Add activity diagram for view monthly

CheeSengg committed:
User guide (#111)

* Updated User Guide

* created new dir for images

* changed isBook to isSold for Seat Class

NicholasLiu97 committed:
update DeveloperGuide.adoc (#112)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

* Updated jar version
Updated user guide
- tips and notes
- content page for navigation
- highlighting of commands and parameters
- minor grammar errors

* Update DG.adoc halfway
- added contents page
- added implementations section

* removed line

* Update DeveloperGuide.adoc
- Formatting

* updated link for images in UserGuide.adoc

NicholasLiu97 committed:
Edited UserGuide.adoc (#113)

* resolved content page issue in UserGuide.adoc
Standardise format or certain sections

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Add images to UserGuide.adoc

NicholasLiu97 committed:
Delete image

NicholasLiu97 committed:
Delete UG_ViewMonthly_GUI.PNG

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Edited implementations in DeveloperGuide.adoc

NicholasLiu97 committed:
update dg

OungKennedy committed:
Formatted dev guide use-case section (#114)

* used Markdown to make use case section more readable.

* fixed formatting error for numbering system.

* fixed indentation issue.

* small improvements to dev guide.

CheeSengg committed:
Updated OptixDateFormatter to handle yr error (#115)

CheeSengg committed:
updated DevGuide, Section 3.2 Sell Seat Implementation (#116)

* updated DevGuide, Section 3.2 Sell Seat Implementation

* updated ViewMonthlyTest

NicholasLiu97 committed:
Updated implementations in DevGuide (#117)

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Update size of images

in Implementations

NicholasLiu97 committed:
Delete UG-remove_alias-success.png

NicholasLiu97 committed:
Delete UG_add-alias_failure.png

NicholasLiu97 committed:
Delete UG_add-alias_illegal.png

NicholasLiu97 committed:
Delete UG_add-alias_success.png

NicholasLiu97 committed:
Delete UG_list-alias_success.png

NicholasLiu97 committed:
Delete UG_reset-alias_success.png

OungKennedy committed:
Changes to DG (#118)

* added junit tests for add, delete edit commands.

* added junit test for list list-date command.

* resolved conflicts.

* updated user guide. Improved formatting to be clearer using MarkDown.

* user guide update for alias related commands.

* used Markdown to make use case section more readable.

* fixed formatting error for numbering system.

* fixed indentation issue.

* small improvements to dev guide.

* added use cases from gdocs.

CheeSengg committed:
Changed Postpone to Reschdule, Bug Fix

OungKennedy committed:
Fixed command alias bug (#136)

* added use cases from gdocs.

* fixed bugs in command aliasing.

NicholasLiu97 committed:
bug fixes (#138)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

* Updated jar version
Updated user guide
- tips and notes
- content page for navigation
- highlighting of commands and parameters
- minor grammar errors

* Update DG.adoc halfway
- added contents page
- added implementations section

* removed line

* Update DeveloperGuide.adoc
- Formatting

* updated link for images in UserGuide.adoc

* resolved content page issue in UserGuide.adoc
Standardise format or certain sections

* Updated implementations in DevGuide

* Corrected use cases

* fixed DeleteCommand bug
- trim show name

* edited message of ViewCommand to show the seat tiering

* improved user interaction for SellSeat command

* updated JUnit test for SellSeat Command

CheeSengg committed:
Bug fix (#139)

* Changed Postpone to Reschdule, Bug Fix

* BugFix AddAliasCommand, SeatDisplay, TabCommand

* Updated DevGuide Design Section

* Fixed spelling error for DG

CheeSengg committed:
Updated devGuide

OungKennedy committed:
Added logging (#141)

* added junit tests for add, delete edit commands.

* added junit test for list list-date command.

* resolved conflicts.

* updated user guide. Improved formatting to be clearer using MarkDown.

* user guide update for alias related commands.

* used Markdown to make use case section more readable.

* fixed formatting error for numbering system.

* fixed indentation issue.

* small improvements to dev guide.

* added use cases from gdocs.

* fixed bugs in command aliasing.

* merged with master. fixed unicode in jar file.

* wrote test for parser.

* added logging feature.

* made small changes to logging functions. this commit passes checkstyle and junit tests.

* made change to logger configurations.

* removed log files.

CheeSengg committed:
Manual Testing Section in DG (#145)

NicholasLiu97 committed:
Updated DG (#146)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

* Updated jar version
Updated user guide
- tips and notes
- content page for navigation
- highlighting of commands and parameters
- minor grammar errors

* Update DG.adoc halfway
- added contents page
- added implementations section

* removed line

* Update DeveloperGuide.adoc
- Formatting

* updated link for images in UserGuide.adoc

* resolved content page issue in UserGuide.adoc
Standardise format or certain sections

* Updated implementations in DevGuide

* Corrected use cases

* fixed DeleteCommand bug
- trim show name

* edited message of ViewCommand to show the seat tiering

* improved user interaction for SellSeat command

* updated JUnit test for SellSeat Command

* minor edit to build.gradle

* updated tab command to reflect output in chat box

* updated Junit test for TabCommand and ShowMap

* updated user guide for add command

* removed "show" from parser

* added manual testing in DG

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Update AboutUs.adoc

NicholasLiu97 committed:
Delete UI.png

NicholasLiu97 committed:
Update AboutUs.adoc

NicholasLiu97 committed:
Update README.adoc

NicholasLiu97 committed:
Update AboutUs.adoc

NicholasLiu97 committed:
Update AboutUs.adoc

NicholasLiu97 committed:
Delete Ui.png

NicholasLiu97 committed:
Upload Ui.png

NicholasLiu97 committed:
Update README.adoc

NicholasLiu97 committed:
Update UserGuide.adoc

OungKennedy committed:
Edited DG  (#147)

* added junit tests for add, delete edit commands.

* added junit test for list list-date command.

* resolved conflicts.

* updated user guide. Improved formatting to be clearer using MarkDown.

* user guide update for alias related commands.

* used Markdown to make use case section more readable.

* fixed formatting error for numbering system.

* fixed indentation issue.

* small improvements to dev guide.

* added use cases from gdocs.

* fixed bugs in command aliasing.

* merged with master. fixed unicode in jar file.

* wrote test for parser.

* added logging feature.

* made small changes to logging functions. this commit passes checkstyle and junit tests.

* made change to logger configurations.

* removed log files.

* edited DG to include logging.

NicholasLiu97 committed:
Minor changes to logging and checkstyle (#148)

* added ViewProfitCommand to view the proft of any show

* Added view profit command

Format: view-profit lion-king|5/5/2020

* fixed bug with incorrect parameters for ViewProfitCommand

* Added ViewMonthlyCommand.java
- Able to check monthly profit from past shows

* added ViewMonthlyCommand.java
Format: view-monthly may 2020

* Update build.gradle

* refactored findMonthly into Model.
- Optimised view-monthly for all range of dates

* updated Junit tests for view-monthly command

* Updated jar version
Updated user guide
- tips and notes
- content page for navigation
- highlighting of commands and parameters
- minor grammar errors

* Update DG.adoc halfway
- added contents page
- added implementations section

* removed line

* Update DeveloperGuide.adoc
- Formatting

* updated link for images in UserGuide.adoc

* resolved content page issue in UserGuide.adoc
Standardise format or certain sections

* Updated implementations in DevGuide

* Corrected use cases

* fixed DeleteCommand bug
- trim show name

* edited message of ViewCommand to show the seat tiering

* improved user interaction for SellSeat command

* updated JUnit test for SellSeat Command

* minor edit to build.gradle

* updated tab command to reflect output in chat box

* updated Junit test for TabCommand and ShowMap

* updated user guide for add command

* removed "show" from parser

* added manual testing in DG

* resolved checkstyle error

* removed test show from parser test

* test for travis CI

* restore changes

NicholasLiu97 committed:
Update UserGuide.adoc

NicholasLiu97 committed:
Update ReassignSeatCommandTest.java

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Update DeveloperGuide.adoc

NicholasLiu97 committed:
Delete DG_Model_Structure_Diagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Update README.adoc

NicholasLiu97 committed:
Update DG

* added API
* updated model documentation

NicholasLiu97 committed:
Delete DG_Model_Structure_Diagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Delete DG_Model_Structure_Diagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Delete DG_ViewMonthly_SequenceDiagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Delete DG_ViewMonthly_SequenceDiagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Delete DG_ViewMonthly_ActivityDiagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Delete DG_ViewMonthly_ActivityDiagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Delete DG_ViewMonthly_ActivityDiagram.png

NicholasLiu97 committed:
Add files via upload

NicholasLiu97 committed:
Update DeveloperGuide.adoc

OungKennedy committed:
Changed picture in dg, refactored logging in commands (#150)

* added junit tests for add, delete edit commands.

* added junit test for list list-date command.

* resolved conflicts.

* updated user guide. Improved formatting to be clearer using MarkDown.

* user guide update for alias related commands.

* used Markdown to make use case section more readable.

* fixed formatting error for numbering system.

* fixed indentation issue.

* small improvements to dev guide.

* added use cases from gdocs.

* fixed bugs in command aliasing.

* merged with master. fixed unicode in jar file.

* wrote test for parser.

* added logging feature.

* made small changes to logging functions. this commit passes checkstyle and junit tests.

* made change to logger configurations.

* removed log files.

* edited DG to include logging.

* refactored logging.

* fixed bug in logging.

* this version passes checkstyle and junit tests.

CheeSengg committed:
Added Figure numbers for images, added Links for API (#151)

* Added Figure numbers for images, added Links for API

* Added PPP

* Bug Fix for PPP

* updated PPP images

* updated Photos PPP

TianchangLiao committed:
Add remove-seat and refund-seat command for full and half refund respectively (#152)

* Add remove-seat command

* Add refund-seat command

OungKennedy committed:
Added ppp

CheeSengg committed:
updated UG and bug fix

NicholasLiu97 committed:
Add PPP

TianchangLiao committed:
Fixes to logger for remove-seat and refund-seat command and improved checkstyle compliance (#153)

* Add remove-seat command

* Add refund-seat command

* Bug fixes

* Bug fixes

NicholasLiu97 committed:
Update cheesengg.adoc

CheeSengg committed:
Updated test package name
